DQMOffline/Trigger/plugins/ParticleNetJetTagMonitor.cc

```diff
@@ -0,0 +1,1108 @@
+    selectedJets.push_back(jet);
+    jetPtCorrectedValues.push_back(jet.pt());
+    auto jref = reco::JetBaseRef(reco::PFJetRef(jetHandle, &j - &(*jetHandle)[0]));
+    if (jetPNETScore[jref])
+      jetPNETScoreValues.push_back(jetPNETScore[jref]);
+    else
+      jetPNETScoreValues.push_back(0);
+  }
+  if (njets_ >= 0 and int(selectedJets.size()) < njets_)
+    return;
+  selectionFlowStatus++;
+  selectionFlow->Fill(selectionFlowStatus);
+
+  // count number of jets over minPNETScoreCut
+  if (std::count_if(jetPNETScoreValues.begin(), jetPNETScoreValues.end(), [&](float score) {
+        return score > minPNETScoreCut_;
+      }) < njets_)
+    return;
+  selectionFlowStatus++;
+  selectionFlow->Fill(selectionFlowStatus);
+
+  // sort descending corrected pt order only if jecHandle is valid otherwise keep the current ordering
+  std::vector<size_t> jetPtSortedIndices(jetPtCorrectedValues.size());
+  std::iota(jetPtSortedIndices.begin(), jetPtSortedIndices.end(), 0);
+  if (jecHandle.isValid()) {
+    std::sort(jetPtSortedIndices.begin(), jetPtSortedIndices.end(), [&](const size_t& i1, const size_t& i2) {
+      return jetPtCorrectedValues[i1] > jetPtCorrectedValues[i2];
+    });
+  }
+
+  // sort descending PNET score order
+  std::vector<size_t> jetPNETScoreSortedIndices(jetPNETScoreValues.size());
+  std::iota(jetPNETScoreSortedIndices.begin(), jetPNETScoreSortedIndices.end(), 0);
+  std::sort(jetPNETScoreSortedIndices.begin(),
+            jetPNETScoreSortedIndices.end(),
+            [&](const size_t& i1, const size_t& i2) { return jetPNETScoreValues[i1] > jetPNETScoreValues[i2]; });
+
+  // trigger object matching (for jets)
+  if (requireHLTOfflineJetMatching_) {
+    edm::Handle<reco::JetTagCollection> jetPNETScoreHLTHandle;
+    iEvent.getByToken(jetPNETScoreHLTToken_, jetPNETScoreHLTHandle);
+    if (!jetPNETScoreHLTHandle.isValid()) {
+      edm::LogWarning("ParticleNetJetTagMonitor") << "HLT Jet tags collection not valid, will skip event \n";
+      return;
+    }
+
+    std::vector<float> jetPNETScoreValuesHLT;
+    std::vector<reco::JetBaseRef> jetHLTRefs;
+    for (const auto& jtag : *jetPNETScoreHLTHandle) {
+        jetPNETScoreValuesHLT.push_back(jtag.second);
+	jetHLTRefs.push_back(jtag.first);
+    }
+
+    // sort in PNET score
+    std::vector<size_t> jetPNETScoreSortedIndicesHLT(jetPNETScoreValuesHLT.size());
+    std::iota(jetPNETScoreSortedIndicesHLT.begin(), jetPNETScoreSortedIndicesHLT.end(), 0);
+    std::sort(
+        jetPNETScoreSortedIndicesHLT.begin(),
+        jetPNETScoreSortedIndicesHLT.end(),
+        [&](const size_t& i1, const size_t& i2) { return jetPNETScoreValuesHLT[i1] > jetPNETScoreValuesHLT[i2]; });
+
+    // match reco and hlt objects considering only the first ntrigobjecttomatch jets for both reco and HLT. Each of them must be matched
+    std::vector<int> matched_obj;
+    for (size_t jreco = 0; jreco < ntrigobjecttomatch_; jreco++) {
+      if (jreco >= jetPNETScoreSortedIndices.size())
```

Comment:
@perrotta indeed I added that check about vector size in order to be sure that there is no access to an invalid position. I will also propagate it back to the master branch tomorrow. Since this monitoring is important for pp collision we want to have it online as soon as the data-taking restarts. Hence, I hope this won't stall the approval of this PR

Reply:
@rgerosa thank you for checking.
Yes, please prepare first the PR for the master, and also the backport for 12_5, in both cases by adding also the fix here [below](https://github.com/cms-sw/cmssw/pull/39146#discussion_r957266639)
Once they are merged, we will move forward also this backport PR for 12_4_X